lib/azure/armrest/model/storage_account.rb

```diff
@@ -194,27 +193,38 @@ def directory_metadata(share, directory, key = access_key, options = {})
       # * marker
       # * maxresults
       # * timeout
+      # * all
+      #
+      # If the :all option is set to true, then this method will collect all
+      # records. Otherwise, it is capped at 5000 records by Azure, or whatever
+      # you set :maxresults to. If you set both the :all option and the :maxresults
+      # option, then all records will be collected in :maxresults batches.
       #
       def files(share, key = access_key, options = {})
         raise ArgumentError, "No access key specified" unless key
 
-        query = {:restype => 'directory', :comp => 'list'}.merge(options).to_query
+        query = "comp=list&restype=directory"
+
+        skip_defs = options[:skip_accessors_definition]
+
+        options.each do |okey, ovalue|
+          unless okey == :skip_accessors_definition
+            query += "&#{okey}=#{[ovalue].flatten.join(',')}"
```

Comment:
Prefer using URI.encode_www_form helper method to ensure things are escaped properly.

```ruby
URI.encode_www_form(okey => ovalue)
```

You could also just build up a big hash and pass it once...

```ruby
query_options = {:restype => 'directory', :comp => 'list'}.merge(options.except(:skip_accessors_definition))
query = URI.encode_www_form(query_options)
```

Reply:
No can do boss man, the `marker` can't be encoded or it blows up.